persistence/pre_persistence/codec.go

```diff
@@ -0,0 +1,54 @@
+package pre_persistence
+
+import (
```

Comment:
Should we add a TODO to move this into shared? Just add it in if so, otherwise I might need to understand more.